packages/react-input/src/components/Input/Input.types.ts

```diff
@@ -1,18 +1,24 @@
+import * as React from 'react';
 import type { ComponentProps, ComponentState, IntrinsicShorthandProps } from '@fluentui/react-utilities';
 
 export type InputSlots = {
   /**
    * Wrapper element which visually appears to be the input and is used for borders, focus styling, etc.
    * (A wrapper is needed to properly position `contentBefore` and `contentAfter` relative to `input`.)
+   *
+   * The root slot receives the `className` and `style` specified directly on the `<Input>`.
+   * All other top-level native props will be applied to the primary slot, `input`.
    */
   root: IntrinsicShorthandProps<'span'>;
 
   /**
    * The actual `<input>` element. `type="text"` will be automatically applied unless overridden.
-   * This is the "primary" slot, so top-level native props (except `className` and `style`) and the
-   * top-level `ref` will go here.
+   *
+   * This is the "primary" slot, so native props specified directly on the `<Input>` will go here
+   * (except `className` and `style`, which go to the `root` slot). The top-level `ref` will
+   * also go here.
    */
-  input: Omit<IntrinsicShorthandProps<'input'>, 'size'>;
+  input: IntrinsicShorthandProps<'input'>;
```

Comment:
IMO it's actually better to omit any native props that have custom handling here, rather than in the `ComponentProps` definition. That way the user can't accidentally specify a prop on the input slot if it won't work as expected.

Reply:
That said, some of the props you're omitting below like `defaultValue`, `value` (and possibly `onChange` pending my comment) don't appear to have special handling in `useInput`. Since you're just adding doc comments, you probably(?) don't need to omit them from the base type.

Reply:
`value` and `defaultValue` need to be omitted because the default definitions include `number | string[]` which is wrong in this case.

Reply:
It becomes an issue to omit the native props here because then you can't set them in `useInput` (and I think this becomes irrelevant if we merge the RFC)